annotation-utils/src/main/java/edu/isi/vista/annotationutils/AceCorpusTextSource.java

```diff
@@ -0,0 +1,52 @@
+package edu.isi.vista.annotationutils;
+
+import com.google.common.base.Charsets;
+import com.google.common.io.Files;
+import edu.isi.nlp.symbols.Symbol;
+import com.google.common.base.Optional;
+import edu.isi.nlp.io.OriginalTextSource;
+import java.io.File;
+import java.nio.file.Path;
+import java.io.IOException;
+
+
+public final class AceCorpusTextSource implements OriginalTextSource{
+    private  File aceDataDirectory;
+    private File[] sourceFileDirectories;
+
+    public Optional<String> getOriginalText(final Symbol docID) throws IOException {
+        String rawSourceText = "";
+        // Search and find file in the source directories
+        for (File sourceDir : sourceFileDirectories){
+            File sourceDocFile = new File(sourceDir, docID.toString() + ".sgm");
+            if(sourceDocFile.exists()) {
+                rawSourceText = Files.asCharSource(sourceDocFile, Charsets.UTF_8).read();
+                break;
+            }
+        }
+        if(rawSourceText.isEmpty()) {
+            throw new IOException("Source file is empty for document: " + docID.toString());
+        }
+        return Optional.of(processAceSourceText(rawSourceText));
+    }
+
+    AceCorpusTextSource(File aceDataDirectory) throws IOException{
+        if(!aceDataDirectory.exists()) {
+            throw new IOException("The aceDataDirectory does not exist");
+        }
+        if(!aceDataDirectory.isDirectory()) {
+            throw new IOException("The aceDataDirectory parameter given is not a directory");
+        }
+        this.aceDataDirectory = aceDataDirectory;
+        this.sourceFileDirectories = aceDataDirectory.listFiles();
+        for(int i = 0; i < this.sourceFileDirectories.length; i++) {
+            this.sourceFileDirectories[i] = new File(sourceFileDirectories[i], "adj");
+        }
+    }
+
+    private String processAceSourceText (String rawSourceText) {
+        return rawSourceText
+                .replaceAll("\\r\\n", "\n")
+                .replaceAll("<[\\s\\S]*?>", "");

```

Comment:
Why not `.*?`?

Reply:
Add a comment explaining to the reader why these replacements are made (ACE offset counting normalizes line endings and does not include markup).

Reply:
Some of the tags (in rare cases) had its opening and closing symbols on different lines which `.*?` does not include, so I decided to use `<[\\s\\S]*?>`

Reply:
Comments added